src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CamundaServiceTest.java

```diff
@@ -251,19 +269,19 @@ void searchWithCriteria_should_succeed() {
                 "someProcessInstanceId"
             );
 
-            Map<String, CamundaVariable> variables = mockVariables();
-
             when(camundaQueryBuilder.createQuery(searchTaskRequest))
                 .thenReturn(camundaSearchQueryMock);
             when(camundaServiceApi.searchWithCriteria(BEARER_SERVICE_TOKEN, camundaSearchQueryMock.getQueries()))
                 .thenReturn(singletonList(camundaTask));
-            when(camundaServiceApi.getVariables(BEARER_SERVICE_TOKEN, camundaTask.getId()))
-                .thenReturn(variables);
+            when(camundaServiceApi.getAllVariables(
+                BEARER_SERVICE_TOKEN,
+                Map.of("variableScopeIdIn", singletonList("someProcessInstanceId"))
+            )).thenReturn(mockedVariablesResponse());
```

Comment:
Not sure if I've missed them elsewhere but probably want some unhappy cases - E.g. if empty response returned?  Also if we have different datasets coming back like when searching for multiple Tasks belonging to different processes maybe?

Reply:
You're right, good spot we need to cover those scenarios I will add more unit tests thanks :)